department-of-veterans-affairs/caseflow
# Fix case detail page crash for nod date timeline updates

### Description
- Fix case details page not loading if NOD date update existed in timeline
* Add separate feature toggle for viewing of NOD date updates
* Add serializer for nod date update
* Add reducer and action for edit nod date update

### Acceptance Criteria
- [ ] Case details page loads when nod date update present
- [ ] Nod date update appears in timeline after patch request sent without page reload
- [ ] Only active tasks appear in `TaskSnapshot`
- [ ] Timeline events sort in proper order in Case Timeline

### Testing Plan
1. Enable feature toggle for `view_nod_date_updates`
2. Go to appeal 
3. Verify that page loads
4. Verify that extra gray line does not appear under active tasks 
5. Add an nod date update
6. Verify the update appears in the timeline

### User Facing Changes
 BEFORE|AFTER
 ---|---
<img width="1420" alt="Screen Shot 2021-02-09 at 4 47 03 PM" src="https://user-images.githubusercontent.com/68879427/107448593-cf754400-6af6-11eb-93ab-69d7ce92ad83.png">|<img width="1039" alt="Screen Shot 2021-02-12 at 9 24 43 PM" src="https://user-images.githubusercontent.com/68879427/107842531-d57a5780-6d78-11eb-9c58-32ac4de730a4.png">
